src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishAnAotApp.cs

```diff
@@ -18,6 +18,31 @@ public GivenThatWeWantToPublishAnAotApp(ITestOutputHelper log) : base(log)
         {
         }
 
+        [Fact]
+        public void Can_Publish_SingleFile_And_NativeAot()
+        {
+            var targetFramework = ToolsetInfo.CurrentTargetFramework;
+            var project = CreateHelloWorldTestProject(targetFramework, "SingleFileAndNativeAot", true);
+            var testAsset = _testAssetsManager.CreateTestProject(project);
+            var buildProperties = project.GetPropertyValues(testAsset.TestRoot, targetFramework);
+            project.RecordProperties("NETCoreSdkPortableRuntimeIdentifier");
+            project.AdditionalProperties["PublishSingleFile"] = "true";
+            project.AdditionalProperties["PublishAot"] = "true";
+
+            var rid = buildProperties["NETCoreSdkPortableRuntimeIdentifier"];
+            var publishCommand = new PublishCommand(testAsset);
+            publishCommand
+                .Execute()
+                .Should().Pass();
+
+            var publishDirectory = publishCommand.GetOutputDirectory(targetFramework, runtimeIdentifier: rid).FullName;
+            var publishedExe = Path.Combine(publishDirectory, $"{project.Name}{Constants.ExeSuffix}");
+
+            var command = new RunExeCommand(Log, publishedExe)
+                .Execute().Should().Pass()
+                .And.HaveStdOutContaining("Hello World");
+        }
+
         [RequiresMSBuildVersionTheory("17.8.0")]
         [MemberData(nameof(Net7Plus), MemberType = typeof(PublishTestUtils))]
```

Comment:
Is there a situation where we run the unit tests on different versions? I don't really understand how we would expect that to work.

Reply:
Reason one - consistency - if the other test has it, then why is this one different?

Reason two - my understanding is that to maintain SDK back compat, we can run the unit tests against older TFMs to test for it - and thus tests which can only work on a given TFM need to be marked accordingly.

But I know next to nothing about this.

Reply:
@rainersigwald Could you shed some light on this? When should we be using `RequiresMSBuildVersion`? These tests are core-SDK only, so I don't think desktop MSBuild is even a consideration.

Reply:
AFAIK `RequiresMSBuildVersion` means "when building using `msbuild.exe` this uses features that are new to the named version". Since `msbuild.exe` is delivered to the test runners through their VS install, it's generally not up to date with the version embedded in the SDK--but Microsoft-internal folks generally _are_, if they're running VS Internal Preview. So that gets some coverage + lights up coverage automatically after the VS on the image updates.

In general, if the PR tests pass I don't think you need the attribute

> These tests are core-SDK only, so I don't think desktop MSBuild is even a consideration.

Really? Based on looking at the test I would have expected it to work driven by either build engine.

Reply:
They probably work fine, I just don't particularly want to support desktop MSBuild if I don't have to. Reduces version swing.